### DISCUSSION TITLE
React Router v7 Docs - Missing a lot of what I used when migrating to v6

### DISCUSSION THREAD

**Discussion by kelsin** [remix-run/react-router#12370] [Q&A]
The main website mentions that I should be able to upgrade to v7 without any breaking changes and continue to use it. I just recently upgraded from v5 to v6 and in doing so I did EVERY recommended upgrade (with all future flags) which meant using route objects at the top level of my app with `createBrowserRouter` and `RouterProvider`.

Now however the docs don't seem to mention this stuff at all? The only provided docs talk about either the full Remix+Framework model, or some VERY simple pages about how to use the Component versions. (The only reference in v7 docs I can find is the bare bones API reference).

Was the messaging in the v6 docs accurate? Even if I don't want to switch my SPA app to the full v7 framework, is it still recommended to use `RouterProvider`?

**Comment by sergiodxa** [comment]
The docs focus on guides, for the API reference to know what's exported and how it's used, there's a link on the header named API Reference that takes you to https://api.reactrouter.com/v7/

**Reply by kelsin** [reply]
Yes, I know about the api reference, but it's not clear how to use `createBrowserRouter` and how to create a routes object from those docs at all. You can get the bare minimum of the shape of arguments and can take a guess at what the fields of the RouteObject (for example) do, but there is no text description or prose about any of it. There is no way I would learn how to use these methods (or that I even should) if not for prior knowledge and the v6 docs:

https://api.reactrouter.com/v7/functions/react_router.createBrowserRouter.html
https://api.reactrouter.com/v7/types/react_router.RouteObject.html
https://api.reactrouter.com/v7/interfaces/react_router.IndexRouteObject.html

This seems at odds with how much the v6 docs were pushing for this method. I know v7 is super new, but I just spent all this time moving my whole app to a method that seems left out entirely of all v7 docs.

**Comment by kelsin** [comment]
I'm sorry if this discussion is in the wrong place. Currently if you go to https://github.com/remix-run/react-router/issues and click on "New Issue" and then "🤔 Usage Question (Github Discussions)" you end up creating an issue in the discussion forum for `remix-run/remix` and not `remix-run/react-router`.

**Reply by kelsin** [reply]
This has been moved, Thanks. The link on https://github.com/remix-run/react-router/issues/new/choose is still wrong.

**Comment by TinsFox** [comment]
The v7 documentation defaults to you having pre-knowledge of v6😭, which is so frustrating.

**Reply by kelsin** [reply]
Yep. I was very lucky that I upgraded from 5 to 6 the week before. If I had delayed that upgrade I would have been VERY lost.
